### DISCUSSION TITLE
Database shells configuration seems broken

### DISCUSSION THREAD

**Discussion by fuzl-llc** [OrchardCMS/OrchardCore#10414] [General]
I've been trying to use database shells and even a very simple setup ends in an exception for me. Here are my steps:

1. Created local MariaDB database called "shells."

2. Created folder and ran:
dotnet new occms

3. Updated appsettings.json with:
  "OrchardCore_Shells_Database": {
    "DatabaseProvider": "MySql",
    "ConnectionString": "server=localhost; database=shells; username=root; password=the-real-pw",
    "TablePrefix": "_orchard",
    "MigrateFromFiles": true
  }

4. Udpated Startup.cs from:
services.AddOrchardCms();
to:
services.AddOrchardCms().AddDatabaseShellsConfiguration();

5. Ran site and get this:

An unhandled exception occurred while processing the request.
ArgumentNullException: The 'OrchardCore.Shells.Database' configuration section should define a 'DatabaseProvider' (Parameter 'DatabaseProvider')
OrchardCore.Shells.Database.Extensions.DatabaseShellContextFactoryExtensions.GetDatabaseContextAsync(IShellContextFactory shellContextFactory, DatabaseShellsStorageOptions options)

Note: I've seen Orchard docs saying to use "OrchardCore_Shells_Database" and to use "OrchardCore.Shells.Database" but I get the same error either way. I also tried using Sqllite for the database provider. I'm using the latest .NET SDK and the 1.0 release of Orchard and the latest code gen templates. Can anyone confirm this is a bug or am I doing something wrong? Thanks!

**Comment by jtkech** [comment]
In your apsettings.json, is it under the `OrchardCore` section?

    "OrchardCore": {
      ...
    }

**Comment by fuzl-llc** [comment]
Thanks for your response! Actually, good question - it wasn't at the time I wrote the above but I fixed it and was still seeing the same issue. On further testing now, I get the same issue only if I specify OrchardCore.Shells.Database with dots separating the parts but OrchardCore_Shells_Database gives a different error:

An unhandled exception occurred while processing the request.
NullReferenceException: Object reference not set to an instance of an object.
OrchardCore.ContentManagement.Display.ContentItemDisplayManager.BuildDisplayAsync(ContentItem contentItem, IUpdateModel updater, string displayType, string groupId)

When I use OrchardCore.Shells.Database I still get the original error even when under the root "OrchardCore" config section.

Even weirder, in the OrchardCore_Shells_Database case, I can just add a /admin to URL and it goes to a login form. When I log in, I'm seeing a totally different orchard site I'm developing on this same PC. It is really weird but seems I've messed something up pretty bad. The App_Data folder is created and there's no appsettings.json in the root App_Data folder or under the /Sites/Default folder. I don't know where the database connection info is coming from.

**Comment by fuzl-llc** [comment]
Well, it seems I had an old environment variable from when I tried to do this before. Still confirming but looks like info was being saved in that old DB.

**Reply by jtkech** [reply]
Use `OrchardCore_Shells_Database`

What I can say is that

These options define the database connection to store the shells settings (in place of using files under App_Data)

They don't define the database connection of the actual site that was specified on the 1st setup, the default being `Sqlite` that uses a `yessql.db` file under App_Data/Sites/TenantName/

But you can use the same connection for both

**Comment by fuzl-llc** [comment]
That was it. I had a an environment variable (I deleted it but I think it was "OrchardCore__OrchardCore_Shells_Database__ConnectionString") that was overriding the "OrchardCore/OrchardCore_Shells_Database/ConnectionString" config setting in the appsettings.json. So, whatever I did in the app settings was being overridden and weirdness ensued. As usual, the computer did EXACTLY what I told it too. Thanks @jtkech for your help!

**Reply by jtkech** [reply]
Okay cool ;)
